### PR TITLE
fastmod: init at 0.4.0

### PR DESCRIPTION
**Pull request by jduan**
###### Motivation for this change

This adds a package for https://github.com/facebookincubator/fastmod, a CLI tool for refactoring code.

###### Things done
I ran `nix-build -A fastmod default.nix` and tested the binary built. It worked as expected.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/fastmod/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ stdenv
+, fetchFromGitHub
+, rustPlatform
+, Security
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "fastmod";
+  version = "v0.4.0";
```
Comment (danieldk) [inline pkgs/tools/text/fastmod/default.nix:9]:
```suggestion
  version = "0.4.0";
```

File: pkgs/tools/text/fastmod/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchFromGitHub {
+    owner = "facebookincubator";
+    repo = pname;
+    rev = version;
```
Comment (danieldk) [inline pkgs/tools/text/fastmod/default.nix:14]:
```suggestion
    rev = "v${version}";
```

File: pkgs/tools/text/fastmod/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 = "0089a17h0wgan3fs6x1la35lzjs1pib7p81wqkh3zcwvx8ffa8z8";
+  };
+
+  cargoSha256 = "0i5i9hsqynb7c1wv63p7y9k7yfl7iz67favrml97gy8l0dzi0666";
+
+  buildInputs = (stdenv.lib.optional stdenv.isDarwin Security);
```
Comment (danieldk) [inline pkgs/tools/text/fastmod/default.nix:20]:
```suggestion
  buildInputs = stdenv.lib.optional stdenv.isDarwin Security;
```

File: pkgs/tools/text/fastmod/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = with stdenv.lib; {
+    description = "A utility that makes sweeping changes to large, shared code bases";
+    homepage = "https://github.com/facebookincubator/fastmod";
+    license = "Apache-2.0";
```
Comment (danieldk) [inline pkgs/tools/text/fastmod/default.nix:25]:
```suggestion
    license = licenses.asl20;
```

File: pkgs/tools/text/fastmod/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    maintainers = with maintainers; [ swolchok ];
```
Comment (danieldk) [inline pkgs/tools/text/fastmod/default.nix:26]: Add a separate commit (ordered before the commit that adds this derivation) to add yourself to `maintainers/maintainer-list.nix`.

**Review by danieldk — CHANGES_REQUESTED** [review]
Thanks for submitting this package and contributing to nixpkgs :+1: !

I have suggested some changes, which should also fix the ofborg build.

File: pkgs/tools/text/fastmod/default.nix (hunk 5 above)
Comment (jduan) [reply pkgs/tools/text/fastmod/default.nix]: @danieldk thanks for your quick response and suggestions. I've applied them and added myself as a maintainer. All CI checks passed. PTAL.

Comment (danieldk) [reply pkgs/tools/text/fastmod/default.nix]: Thanks for making these changes! Could you change the first line of the message of the commit where you add yourself to `maintainers.nix` to:

```
maintainers: add jduan
```

And then squash all the other commits (the ones for the derivation itself) into one commit with the following initial line in the commit message:

```
fastmod: init at 0.4.0
```

All commit messages in `nixpkgs` follow the same patterns to make it easy to find what was changed when ;).

In the commit history, the commit that adds you to `maintainers.nix` should be ordered before the one where the derivation is added.

If you need help fixing up the history, you can enable *Allow edits and access to secrets by maintainers* on in the rightmost column and let me know, and I can do the cleanup.

Comment (jduan) [reply pkgs/tools/text/fastmod/default.nix]: @danieldk thanks for suggesting the convention used by this repo! I've changed the commit history as you suggested. PTAL. Again, thanks for making my first contribution to this repo a pleasant experience!

File: pkgs/tools/text/fastmod/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  version = "0.4.0";
+
+  src = fetchFromGitHub {
+    owner = "facebookincubator";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0089a17h0wgan3fs6x1la35lzjs1pib7p81wqkh3zcwvx8ffa8z8";
+  };
+
+  cargoSha256 = "0i5i9hsqynb7c1wv63p7y9k7yfl7iz67favrml97gy8l0dzi0666";
```
Comment (danieldk) [inline pkgs/tools/text/fastmod/default.nix:18]: Thanks for making all the changes and fixing up the git history. It seems that the `cargoSha256` hash is still wrong, leading to the ofborg failure (I get the same error on my machine). With the following change it should be fine:

```suggestion
  cargoSha256 = "02nkxjwfiljndmi0pv98chfsw9vmjzgmp5r14mchpayp4943qk9m";
```

**Review by danieldk — CHANGES_REQUESTED** [review]
Sorry, one more glitch and then it should be good to go!

File: pkgs/tools/text/fastmod/default.nix (hunk 6 above)
Comment (jduan) [reply pkgs/tools/text/fastmod/default.nix]: Sorry, I should've tried "nix-build" after making those changes. I've accepted your suggestion and squashed the changes again. PTAL.

PS: nix-build worked for me locally.

**Review by danieldk — APPROVED** [review]
LGTM. Thanks a lot!

Result of `nixpkgs-review pr 91196` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- fastmod
</details>

Tested the resulting binary.

**Comment by jduan** [comment]
@danieldk thanks for merging this PR and guiding me along the way. Really appreciate your time!

**Pull request merged**